dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,203 @@
+# Minimal OS for Fobnail project
+
+The Fobnail project aims to provide a reference architecture for building
+offline integrity measurement servers on the USB device and clients running in
+Dynamically Launched Measured Environments (DLME). It should allow the Fobnail
+owner to verify the trustworthiness of the running system before performing any
+sensitive operation.
+
+According to the information provided at TrenchBoot Summit 2021 (the fragment
+discussed here - https://youtu.be/xZoCtNV8Qs0?t=5062), the minimum OS, together
+with the connected Fobnail Token, should allow determining the state of the
+platform before the target OS is launched. As we can see in the presentation,
+`heads` would be responsible for the platform's trust control, while in this
+document, we want to present the results of research on the possibilities of
+using various operating systems.
+
+## Different OSs propositions
+
+The research effect is presented below. 4 systems were considered:
+
+* Zephyr
+* Xous
+* seL4
+* Linux
+
+Each of them has a short description, an analysis of the launch in DLME, and the
+possibilities and potential problems that will have to be addressed for the
+Fobnail Token to be functional.
+
+### Zephyr
```

Comment:
It would be way better to have table based comparison in light of all known Fobnail Project requirements. Ideally we should say about requirements in short here, but there should be documentation which explain each requirement in more details. Those requirements should be user visible features not developers requirements. @macpijan cc

Reply:
The breakdown of this report is as follows. In the introduction (section OS requirements), we indicate the hard and soft req that we take into account. Then we briefly describe each OS in a separate section. Finally, we summarize once again all the requirements in the table.

Reply:
@TomaszAIR yes, after recent improvements' layout is way better, so I will resolve that and wait until @dpsmith suggestions would be implemented.